pkg/internal/parser.go

```diff
@@ -162,37 +152,276 @@ func sanitize(groupName string) string {
+				a.ResultType = fmt.Sprintf("*%sapi.%s", strings.ToLower(name), name)
+			} else {
+				a.ResultType = fmt.Sprintf("*%sapi%s.%s", a.PkgName, a.Version, name)
+			}
+
+		}
+
+		if extension.Input != nil {
+			name, pkg := getPkgType(*extension.Result)
+			if len(pkg) > 0 {
+				*a.importList = append(*a.importList, util.ImportFormat(fmt.Sprintf("%sapi", strings.ToLower(name)), pkg))
+				a.InputType = fmt.Sprintf("*%sapi.%s", strings.ToLower(name), name)
+				if *extension.Verb == "apply" {
+					_, gvString := codegenutil.ParsePathGroupVersion(pkg)
+					*a.importList = append(*a.importList, util.ImportFormat(fmt.Sprintf("%sapplyconfig", strings.ToLower(name)), fmt.Sprintf("%s/%s", a.ApplyConfigurationPackage, gvString)))
+					a.InputType = fmt.Sprintf("*%sapplyconfig.%sApplyConfiguration", strings.ToLower(name), name)
+				}
+			} else {
+				a.InputType = fmt.Sprintf("*%sapi%s.%s", a.PkgName, a.Version, name)
+				if *extension.Verb == "apply" {
+					a.InputType = fmt.Sprintf("*%sapply%s.%sApplyConfiguration", a.PkgName, a.Version, name)
+				}
+			}
+			a.InputName = strings.ToLower(name)
+		}
+
+		if *extension.Verb == "get" {
+			a.Method = *extension.Method
+			if err := templateExecute(getTemplate, *a); err != nil {
+				return err
+			}
+		}
+
+		if *extension.Verb == "create" {
+			a.Method = *extension.Method
+			adjTemplate := adjustTemplate(createTemplate, "create")
+			if err := templateExecute(adjTemplate, *a); err != nil {
+				return err
+			}
+		}
+
+		if *extension.Verb == "update" {
+			a.Method = *extension.Method
+			adjTemplate := adjustTemplate(updateTemplate, "update")
+			if err := templateExecute(adjTemplate, *a); err != nil {
+				return err
+			}
+		}
+
+		if *extension.Verb == "apply" {
+			a.Method = *extension.Method
+			adjTemplate := adjustTemplate(applyTemplate, "apply")
+			if err := templateExecute(adjTemplate, *a); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// Execute template and append the contents to the writer.
+func templateExecute(templateName string, data api) error {
+	funcMap := template.FuncMap{
+		"value": util.Value,
+		"typepkg": func(a api) string {
+			return fmt.Sprintf("*%sapi%s.%s", a.PkgName, a.Version, a.Name)
+		},
+	}
+	templ, err := template.New("wrapper").Funcs(funcMap).Parse(templateName)
 	if err != nil {
 		return err
 	}
-	return templ.Execute(a.writer, a)
+	return templ.Execute(data.writer, data)
+}
+
+// hasVerb returns true if the verb is to be scaffolded,
+// if it is a part of "skipVerbs" then it returns false.
+func (a *api) hasVerb(verb string) bool {
+	if len(a.SkipVerbs) == 0 {
+		return true
+	}
+	for _, s := range a.SkipVerbs {
+		if verb == s {
+			return false
+		}
+	}
+	return true
+}
+
+// getPkgType returns the result override package path and the type.
+func getPkgType(input string) (string, string) {
+	parts := strings.Split(input, ".")
+	return parts[len(parts)-1], strings.Join(parts[0:len(parts)-1], ".")
```

Comment:
If the input is something like `A.B.C.D`, I would like to have the output as `D`, `A.B.C`. Since I am not preemptively aware of the number of separators in input, didn't use `SplitN` here.